python/cuml/test/test_arima.py

```diff
@@ -14,543 +14,432 @@
+#                                  Test data                                  #
+###############################################################################
+
+# Common structure to hold the data, the reference and the testing parameters
+ARIMAData = namedtuple('ARIMAData', ['batch_size', 'n_obs', 'dataset', 'start',
+                                     'end', 'tolerance_integration'])
+
+# ARIMA(2,0,0)
+test_200 = ARIMAData(
+    batch_size=8,
+    n_obs=15,
+    dataset="long_term_arrivals_by_citizenship",
+    start=10,
+    end=25,
+    tolerance_integration=0.001
+)
+
+# ARIMA(0,0,2) with intercept
+test_002c = ARIMAData(
+    batch_size=7,
+    n_obs=20,
+    dataset="net_migrations_auckland_by_age",
+    start=15,
+    end=30,
+    tolerance_integration=0.001
+)
+
+# ARIMA(0,1,0) with intercept
+test_010c = ARIMAData(
+    batch_size=4,
+    n_obs=17,
+    dataset="cattle",
+    start=10,
+    end=25,
+    tolerance_integration=0.001
+)
+
+# ARIMA(1,1,0)
+test_110 = ARIMAData(
+    batch_size=1,
+    n_obs=137,
+    dataset="police_recorded_crime",
+    start=100,
+    end=150,
+    tolerance_integration=0.001
+)
+
+# ARIMA(0,1,1)
+test_011 = ARIMAData(
+    batch_size=16,
+    n_obs=28,
+    dataset="deaths_by_region",
+    start=20,
+    end=40,
+    tolerance_integration=0.005
+)
+
+# ARIMA(1,2,1)
+test_121 = ARIMAData(
+    batch_size=2,
+    n_obs=137,
+    dataset="population_estimate",
+    start=100,
+    end=150,
+    tolerance_integration=0.05
+)
+
+# ARIMA(1,0,1)(1,1,1)_4
+test_101_111_4 = ARIMAData(
+    batch_size=3,
+    n_obs=101,
+    dataset="alcohol",
+    start=80,
+    end=110,
+    tolerance_integration=0.005
+)
+
+# ARIMA(1,1,1)(2,0,0)_4
+test_111_200_4 = ARIMAData(
+    batch_size=14,
+    n_obs=123,
+    dataset="hourly_earnings_by_industry",
+    start=115,
+    end=130,
+    tolerance_integration=0.05
+)
+
+# ARIMA(1,1,2)(0,1,2)_4
+test_112_012_4 = ARIMAData(
+    batch_size=2,
+    n_obs=179,
+    dataset="passenger_movements",
+    start=160,
+    end=200,
+    tolerance_integration=0.001
+)
+
+# ARIMA(1,1,1)(1,1,1)_12
+test_111_111_12 = ARIMAData(
+    batch_size=12,
+    n_obs=279,
+    dataset="guest_nights_by_region",
+    start=260,
+    end=290,
+    tolerance_integration=0.001
+)
+
+# Dictionary matching a test case to a tuple of model parameters
+# (a test case could be used with different models)
+# (p, d, q, P, D, Q, s, k) -> ARIMAData
+test_data = {
+    (2, 0, 0, 0, 0, 0, 0, 0): test_200,
+    (0, 0, 2, 0, 0, 0, 0, 1): test_002c,
+    (0, 1, 0, 0, 0, 0, 0, 1): test_010c,
+    (1, 1, 0, 0, 0, 0, 0, 0): test_110,
+    (0, 1, 1, 0, 0, 0, 0, 0): test_011,
+    (1, 2, 1, 0, 0, 0, 0, 0): test_121,
+    (1, 0, 1, 1, 1, 1, 4, 0): test_101_111_4,
+    (1, 1, 1, 2, 0, 0, 4, 0): test_111_200_4,
+    (1, 1, 2, 0, 1, 2, 4, 0): test_112_012_4,
+    (1, 1, 1, 1, 1, 1, 12, 0): test_111_111_12,
+}
+
+# TODO: test greater orders! (at least 3, 4)
+# TODO: test greater seasonal periods! (52)
```

Comment:
Please file separate issues for these so that we don't miss them

Reply:
#1648